### PR TITLE
Feature/new ux fever

### PR DESCRIPTION
**Pull request by mellowgeek**

### CODE REVIEW

**Comment by mellowgeek** [comment]
Switching between Fahrenheit and Celsius is not implemented in the Highest Temperature screen yet.

**Comment by mellowgeek** [comment]
Screenshot_1588977397

Screenshot_1588977405

Screenshot_1588977409

Screenshot_1588977416

Screenshot_1588977431

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                android:orientation="horizontal"
+                android:layout_marginTop="38dp"
+                android:gravity="center">
+
+                <androidx.constraintlayout.widget.ConstraintLayout
+                    android:layout_width="126dp"
+                    android:layout_height="126dp"
+                    android:layout_marginEnd="46dp"
+                    android:background="@drawable/circle_button_background"
+                    android:onClick="@{() -> vm.onClickArmpit()}">
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_fever_temperature_spot.xml:118]: For what do we need these wrappers? Could all the attributes be in the `TextView`s maybe? If needed, `FrameLayout` seems a better fit.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        <LinearLayout
+            android:id="@+id/options"
+            android:orientation="vertical"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="44dp"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/headline">
+        <LinearLayout
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_fever_temperature_spot.xml:51]: There's a lot of nesting. It should be possible to show the 4 buttons directly in a constraint layout.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                android:layout_height="wrap_content"
+                android:orientation="horizontal"
+                android:layout_marginTop="38dp"
+                android:gravity="center">
+
+                <androidx.constraintlayout.widget.ConstraintLayout
+                    android:layout_width="126dp"
+                    android:layout_height="126dp"
+                    android:layout_marginEnd="46dp"
+                    android:background="@drawable/circle_button_background"
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_fever_temperature_spot.xml:117]: I'd prefer if these buttons are in `RecyclerView`, but let's keep it like this for now. We can refactor the UI later. But at least create reusable style(s) / dimensions, so we don't have to edit every button if we want to e.g. change the size or the background drawable.

File: app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        symptomFlowManager.onBack()
+    }
+
+    fun onBackPressed() {
+        onBack()
+        navigation.navigate(Back)
+    }
+
+    private fun cToF(c: Double) = (9/5.0 * c) + 32
+    private fun fToC(f: Double) = 5/9.0 * (f - 32)
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt:46]: What about something like this:

```kotlin
sealed class Temperature {
    data class Celsius(val value: Float) : Temperature() {
        override fun toFarenheit(): Farenheit =
            Farenheit(value) // TODO convert
    }
    data class Farenheit(val value: Float) : Temperature() {
        override fun toCelsius(): Celsius =
            Celsius(value) // TODO convert
    }

    open fun toCelsius(): Celsius = when (this) {
        is Celsius -> this
        is Farenheit -> toCelsius()
    }

    open fun toFarenheit(): Farenheit = when (this) {
        is Farenheit -> this
        is Celsius -> toFarenheit()
    }
}
```

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml (hunk 2 above)
Comment (mellowgeek) [reply app/src/main/res/layout/fragment_fever_temperature_spot.xml]: Reduced nesting and put everything into the constraint layout.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml (hunk 1 above)
Comment (mellowgeek) [reply app/src/main/res/layout/fragment_fever_temperature_spot.xml]: Attributes moved to the TextView.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml (hunk 3 above)
Comment (mellowgeek) [reply app/src/main/res/layout/fragment_fever_temperature_spot.xml]: Moved a lot of the dimensions and style elements into styles for the buttons, also did the same for elements reused across the pages such as the headline and the toolbars.

File: app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt (hunk 4 above)
Comment (mellowgeek) [reply app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt]: Wasn't exactly sure how to use this, but put it in place for now, see latest commit. Let me know if that was what you were thinking, or how I should fix it. Thanks!

Comment (i-schuetz) [reply app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt]: Almost! Now replace the old `Temperature` with this (you've to make it `Parcelable` I think). This way always know the unit it's in.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml [hunk 5]
```diff
@@ -12,162 +12,68 @@
...
-                <androidx.constraintlayout.widget.ConstraintLayout
-                    android:layout_width="126dp"
-                    android:layout_height="126dp"
-                    android:background="@drawable/circle_button_background"
-                    android:onClick="@{() -> vm.onClickOther()}">
-
+                        android:layout_marginTop="44dp"
+                        app:layout_constraintTop_toBottomOf="@id/mouth"
+                        app:layout_constraintStart_toStartOf="@id/mouth"
+                        app:layout_constraintEnd_toStartOf="other"
```
Comment (i-schuetz) [inline app/src/main/res/layout/fragment_fever_temperature_spot.xml:60]: This should be `@id/other`

File: app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt (hunk 4 above)
Comment (mellowgeek) [reply app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt]: Gotcha, I moved the class into SymptomInputs and replaced the existing temperature class, so now we can pass in Celsius or Fahrenheit objects from the viewmodel.

File: app/src/main/res/layout/fragment_fever_temperature_spot.xml (hunk 5 above)
Comment (mellowgeek) [reply app/src/main/res/layout/fragment_fever_temperature_spot.xml]: Fixed!

File: app/src/main/java/org/coepi/android/domain/symptomflow/SymptomInputs.kt [hunk 6]
```diff
@@ -65,5 +65,26 @@ data class SymptomInputs(
     }
 }
 
-@Parcelize
-data class Temperature(val value: Float): Parcelable
+sealed class Temperature: Parcelable {
+    @Parcelize
+    data class Celsius(val value: Float) : Temperature() {
+        override fun toFarenheit(): Fahrenheit =
+            Fahrenheit(value) // TODO convert
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/domain/symptomflow/SymptomInputs.kt:72]: Solve the TODOs? 😉

File: app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt (hunk 4 above)
Comment (i-schuetz) [reply app/src/main/java/org/coepi/android/ui/symptoms/fever/FeverHighestTemperatureViewModel.kt]: I'd leave it as a standalone class, it's unlikely that we will use it outside the inputs but it's still independent.

**Comment by i-schuetz** [comment]
Looks good now 👍

**Pull request merged**